### ISSUE TITLE
New startup error after recent updates

### ISSUE THREAD

**Issue by dreimer1986** [flobz/psa_car_controller#91]
**Describe the bug**
No output from server after updating to 41ba304044d0c8791316346d6ed69276a6d97a81 and a new error.

**To Reproduce**
1. what command did you use: python.exe server.py -f config.json -l 192.168.181.50 -c charge_config.json -r
2. when the bug append ? Directly on startup
3. Does it persist after restarting the program? yes

**Config file**
Give the anonymize content of the config file

```
{
    "abrp": {
        "abrp_enable_vin": [],
        "token": ""
    },
    "client_id": "...",
    "client_secret": "...",
    "co2_signal_api": null,
    "country_code": "DE",
    "customer_id": "AP-ACNXX00005711XXX",
    "proxies": {
        "http": "",
        "https": ""
    },
    "realm": "clientsB2CPeugeot",
    "refresh_token": "dc69f927-7295-4a76-aa28-af9a40a2a0da",
    "remote_refresh_token": "18119a72-288a-4058-b46e-a9da3d2123cb",
    "weather_api": "..."
}
```

**Environment (please complete the following information):**
 - OS: Windows 10 Pro x64
 - Android app Version: MYPEUGEOT_v1.29.0.apk
 - Brand and model of car: Peugeot e-208
 - Application version (commit number, branch): 41ba304044d0c8791316346d6ed69276a6d97a81

**Additional context**

```
2021-04-30 22:16:50,186 :: INFO :: server start
2021-04-30 22:16:50,815 :: INFO :: save config change
2021-04-30 22:16:51,413 :: INFO :: ["{'vin': 'XXXX', 'vehicle_id': '1566c497a5655686153316861545651774d4467784e54496a497a67334d546b78597a526a593245794d5445334f57526b5a6a6331597a4d775a6a686a4d47493259325a69595441324e575a6d4f5459304d4445355a5451774f44466a4f54566c4e544a6d4e6d5a695932566a4d54453d', 'label': 'e-208', 'brand': 'Peugeot', 'abrp_name': 'peugeot:e208:20:50', 'battery_power': 46, 'fuel_capacity': 0, 'max_elec_consumption': 70, 'max_fuel_consumption': 0}"]
2021-04-30 22:16:51,727 :: INFO :: save config change
2021-04-30 22:16:52,237 :: INFO :: ask state to XXX
2021-04-30 22:16:52,440 :: INFO :: Connected with result code {'session present': 0}
2021-04-30 22:16:52,441 :: INFO :: subscribe to psa/RemoteServices/to/cid/AP-ACNXX00005711XXX/#
2021-04-30 22:16:52,442 :: INFO :: subscribe to psa/RemoteServices/events/MPHRTServices/XXX
2021-04-30 22:16:52,528 :: INFO :: save config change
2021-04-30 22:16:52,530 :: INFO :: {"access_token": "d4ecc9e3-4582-44ba-a754-c35c8351faf7", "customer_id": "AP-ACNXX00005711XXX", "correlation_id": "8a6ea735d4a349618cbfc23396bfb3ca20210430201652529", "req_date": "2021-04-30T20:16:52Z", "vin": "XXXX", "req_parameters": {"action": "state"}}
2021-04-30 22:16:52,569 :: WARNING :: Can't get language
2021-04-30 22:16:52,778 :: INFO :: mqtt msg psa/RemoteServices/to/cid/AP-ACNXX00005711XXX/VehicleState/ProcessManagement b'{"process_date":"2021-04-30T20:16:53Z","vin":"XXXX","correlation_id":"8a6ea735d4a349618cbfc23396bfb3ca20210430201652529","process_code":"900","process_message":"Your request has been accepted and is being processed"}'
2021-04-30 22:16:52,810 :: INFO :: mqtt msg psa/RemoteServices/to/cid/AP-ACNXX00005711XXX/VehicleState b'{"resp_date":"2021-04-30T20:16:53Z","vin":"XXXX","correlation_id":"8a6ea735d4a349618cbfc23396bfb3ca20210430201652529","return_code":"0","resp_data":{"date":"2021-04-30T20:00:46Z","etat_res_elec":0,"precond_state":{"available":1,"programs":{"program1":{"hour":34,"minute":7,"on":0,"day":[0,0,0,0,0,0,0]},"program2":{"hour":34,"minute":7,"on":0,"day":[0,0,0,0,0,0,0]},"program3":{"hour":34,"minute":7,"on":0,"day":[0,0,0,0,0,0,0]},"program4":{"hour":34,"minute":7,"on":0,"day":[0,0,0,0,0,0,0]}},"asap":0,"status":0,"aff":1},"charging_state":{"program":{"hour":21,"minute":30},"available":1,"remaining_time":0,"rate":0,"cable_detected":0,"soc_batt":56,"autonomy_zev":94,"type":0,"aff":1,"hmi_state":1,"mode":0},"vin":"XXXX","reason":4,"signal_quality":2,"sev_stop_date":"2021-04-30T19:59:30Z","fds":["NDR01","NAE01","NBM01","NCG01","NAO01","NAS01"],"sev_state":0,"obj_counter":11,"privacy_customer":0,"privacy_applicable":0,"privacy_applicable_max":2,"superlock_state":0}}'
2021-04-30 22:16:53,074 :: INFO :: update_data
2021-04-30 22:16:53,077 :: WARNING :: There is 976 to fetch from API, it can take some time
2021-04-30 22:16:53,078 :: WARNING :: Failed to generate figure, there is probably not enough data yet
 * Running on http://192.168.181.50:5000/ (Press CTRL+C to quit)
2021-04-30 22:17:03,510 :: WARNING :: There is 976 to fetch from API, it can take some time
Exception on /get_vehicleinfo/XXX [GET]
Traceback (most recent call last):
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\reime\AppData\Local\Programs\Python\Python39\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\psa_car_controller\web\views.py", line 155, in get_vehicle_info
    response=json.dumps(myp.get_vehicle_info(vin).to_dict(), default=str),
  File "C:\psa_car_controller\my_psacc.py", line 144, in get_vehicle_info
    self.record_info(car)
  File "C:\psa_car_controller\my_psacc.py", line 473, in record_info
    Database.record_position(self.weather_api, car.vin, mileage, latitude, longitude, altitude, date, level,
  File "C:\psa_car_controller\web\db.py", line 211, in record_position
    conn = Database.get_db()
  File "C:\psa_car_controller\web\db.py", line 113, in get_db
    Database.init_db(conn)
  File "C:\psa_car_controller\web\db.py", line 96, in init_db
    Database.add_altitude_to_db(conn)
  File "C:\psa_car_controller\web\db.py", line 171, in add_altitude_to_db
    logger.debug("add altitude for %s positions point", len(nb_null))
TypeError: object of type 'int' has no len()
192.168.181.42 - - [30/Apr/2021 22:17:03] "←[35m←[1mGET /get_vehicleinfo/XXX HTTP/1.1←[0m" 500 -
2021-04-30 22:17:03,515 :: INFO :: <Request 'http://192.168.181.50:5000/charge_control?vin=XXX&%3Falways_check=true' [GET]>
2021-04-30 22:17:03,517 :: INFO :: save config change
192.168.181.42 - - [30/Apr/2021 22:17:03] "←[37mGET /charge_control?vin=XXX&?always_check=true HTTP/1.1←[0m" 200 -
```

**Comment by flobz** [comment]
Thanks for reporting, 
it should be fixed with 8efa4c0.

**Comment by dreimer1986** [comment]
It indeed is. Works fine again. Thanks for the fix! Now back to waiting for the new update from PSA ^^
